kategory-core/src/main/kotlin/kategory/data/Weak.kt

```diff
@@ -0,0 +1,58 @@
+package kategory
+
+import java.lang.ref.WeakReference
```

Comment:
This will not be portable, I have to keep an eye on these because we may have to move them around.